iam/models.py

```diff
@@ -0,0 +1,74 @@
+from flask_sqlalchemy import SQLAlchemy
+
+from . import hasher
+
+
+db = SQLAlchemy()
+
+
+class Organization(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(256), nullable=False)
+
+    def __repr__(self):
+        return f'<{self.__class__.__name__} {self.id}: {self.name}>'
+
+
+class Project(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(256), nullable=False)
+
+    # note: a project should belong to an organization or user, but not both
+    organization_id = db.Column(db.Integer, db.ForeignKey('organization.id'))
+    organization = db.relationship('Organization',
+                                   backref=db.backref('projects'))
+
+    user_id = db.Column(db.Integer, db.ForeignKey('user.id'))
+    user = db.relationship('User', backref=db.backref('projects'))
+
+    def __repr__(self):
+        return f'<{self.__class__.__name__} {self.id}: {self.name}>'
+
+
+class User(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+
+    password = db.Column(db.String(128))
+    refresh_token = db.Column(db.String(64))
+    refresh_token_expiry = db.Column(db.DateTime)
+
+    firebase_uid = db.Column(db.String(256))
+
+    first_name = db.Column(db.String(256))
+    last_name = db.Column(db.String(256))
+    email = db.Column(db.String(256), unique=True, nullable=False)
+
+    organization_id = db.Column(db.Integer, db.ForeignKey('organization.id'))
+    organization = db.relationship('Organization', backref=db.backref('users'))
+
+    def __repr__(self):
+        return (f'<{self.__class__.__name__} {self.id}: {self.full_name} '
+                f'({self.email})>')
+
+    @property
+    def full_name(self):
+        return f'{self.first_name} {self.last_name}'
+
+    def set_password(self, password):
+        if not password:
+            raise ValueError("Password cannot be empty")
+        self.password = hasher.encode(password)
+
+    def check_password(self, password):
+        return hasher.verify(password, self.password)
+
+    @property
+    def claims(self):
+        projects = {p.id for p in self.projects}
+        if self.organization_id is not None:
+            projects = projects.union({
+                p.id for p in self.organization.projects})
```

Comment:
You're right, that would be better and more consistent.

However I suggest that we leave all code style checks like this to the CI QA. If there is a rule you would like to enforce then the first step is to make an addition to our [code style guide](https://github.com/biosustain/biosustain/tree/master/groups/DD-DeCaF/code-style-settings/py). If accepted, then add the new rule to the projects QA configuration. That way we can automate all these checks. And style that is accepted by our QA process should be accepted by all team members.

This allows our code reviews to focus on larger architectual topics, rather than minor style issues that can be checked automatically.

Reply:
alright, I'm fine with this maybe I'll look for a rule for this